### PR TITLE
Add support for deepfashion2 dataset

### PR DESCRIPTION
**Pull request by WangXin93**
## Motivation

Hi, I added several config to enable the framework to train the deepfashion2 dataset. Hope it can help when somebody need to do more complicated fashion landmark tasks.

## Modification

The modification lies at the `configs` folder. a new dataset config `deepfashion2.py` in the `_base_` folder, and several config files in the `configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2` folder for different tasks.

No new class is created.

The description of usage is written in `docs/en/tasks/2d_fashion_landmark.md `.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by jin-s13** [comment]
Thanks for your contribution! The commit looks great. 

We will review it ASAP. 

BTW, I noticed that you have shared some pre-trained models. Would you like to also share the training logs?

**Comment by WangXin93** [comment]
> Thanks for your contribution! The commit looks great.
> 
> We will review it ASAP.
> 
> BTW, I noticed that you have shared some pre-trained models. Would you like to also share the training logs?

Sure. I have update the log url in the most recent commit. 

I am trying to figure out the what caused the Codecov failure. I appreciate if there is any help or clues.

File: docs/en/tasks/2d_fashion_landmark.md [hunk 1]
```diff
@@ -74,3 +75,64 @@ mmpose
...
+├── configs
+`── data
+    │── deepfashion2
+        │── train
+            │-- deepfashion2_short_sleeved_outwear_train.json
+            │-- deepfashion2_short_sleeved_dress_train.json
+            │-- deepfashion2_skirt_train.json
+            │-- deepfashion2_sling_dress_train.json
+            │-- ...
+            │-- img
```
Comment (jin-s13) [inline docs/en/tasks/2d_fashion_landmark.md:118]: The original data structure is:
```
Training images: train/image Training annotations: train/annos
Validation images: validation/image Validation annotations: validation/annos
Test images: test/image
```
How about keep the name of `image`, instead of renaming it to `img`?

Comment (WangXin93) [reply docs/en/tasks/2d_fashion_landmark.md:118]: > The original data structure is:
> 
> ```
> Training images: train/image Training annotations: train/annos
> Validation images: validation/image Validation annotations: validation/annos
> Test images: test/image
> ```
> 
> How about keep the name of `image`, instead of renaming it to `img`?

I will try

File: configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/res50_deepfashion2_trousers_256x192.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    num_joints=channel_cfg['dataset_joints'],
+    dataset_channel=channel_cfg['dataset_channel'],
+    inference_channel=channel_cfg['inference_channel'],
+    soft_nms=False,
+    nms_thr=1.0,
+    oks_thr=0.9,
+    vis_thr=0.2,
+    use_gt_bbox=False,
+    det_bbox_thr=0.0,
+    bbox_file='data/deepfashion2/validation/detection_results/'
```
Comment (jin-s13) [inline configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/res50_deepfashion2_trousers_256x192.py:108]: Could you please also share the detection result files?

File: docs/en/tasks/2d_fashion_landmark.md (hunk 1 above)
Comment (jin-s13) [reply docs/en/tasks/2d_fashion_landmark.md:118]: Maybe it is a typo?

File: configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/res50_deepfashion2_trousers_256x192.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+total_epochs = 210
+log_config = dict(
+    interval=10,
+    hooks=[
+        dict(type='TextLoggerHook'),
+        # dict(type='TensorboardLoggerHook')
+    ])
+
+channel_cfg = dict(
+    num_output_channels=294,
```
Comment (jin-s13) [inline configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/res50_deepfashion2_trousers_256x192.py:28]: If I understand correctly, all these models (for different fashion categories) will produce 294 keypoints (although only part of the keypoints are trained). 
Is my understanding correct?

**Comment by jin-s13** [comment]
> Sure. I have updated the log url in the most recent commit.
> 
> I am trying to figure out the what caused the Codecov failure. I appreciate if there is any help or clues.

Please simply ignore the Codecov failure. Detection result files seem to be missing. 

We plan to follow your instructions to prepare the dataset and train a few models to check the model accuracy. If the accuracy is okay, we will merge this PR.

File: docs/en/tasks/2d_fashion_landmark.md (hunk 1 above)
Comment (WangXin93) [reply docs/en/tasks/2d_fashion_landmark.md:118]: > Maybe it is a typo?

This is my experiment setting problem. I will fix that

File: configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/res50_deepfashion2_trousers_256x192.py (hunk 2 above)
Comment (WangXin93) [reply configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/res50_deepfashion2_trousers_256x192.py]: Hi, I realize that the detection bounding box is never used during training and validation. `DeepfashionDataset` has no code about `bbox_file`. There are ground-truth bounding box annotations in deepfashion2 validation set. This is how I evaluate the performance.

So I change config to the following code for simiplicity. 

```python
use_gt_bbox=True,
bbox_file=None
```

Comment (jin-s13) [reply configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/res50_deepfashion2_trousers_256x192.py]: It is okay to use ground-truth bounding boxes for evaluation.

File: configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/resnet_deepfashion2.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+}
+```
+
+</details>
+
+Results on DeepFashion2 val set
+
+| Set                   |                        Arch                         | Input Size | PCK@0.2 |  AUC  | EPE  |                        ckpt                         |                         log                         |
+| :-------------------- | :-------------------------------------------------: | :--------: | :-----: | :---: | :--: | :-------------------------------------------------: | :-------------------------------------------------: |
+| short_sleeved_shirt   | [pose_resnet_50](/configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/res50_deepfashion2_short_sleeved_shirt_256x192.py) |  256x256   |  0.988  | 0.703 | 10.2 | [ckpt](https://drive.google.com/file/d/1waYaWYPXum9Lncv4qmd0H7lt8Abgb0Ry/view?usp=share_link) | [log](https://drive.google.com/file/d/1RCsyvHDkAiA_SMwLoQ9YUvgq8jrCcmUm/view?usp=share_link) |
```
Comment (jin-s13) [inline configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/resnet_deepfashion2.md:55]: Some links of the log seems to be incorrect. For example, the log link for _short_sleeved_shirt_ is the same as that for _long_sleeved_shirt_ .

Copy-pasting the link is error-prone. How about organizing these files into a directory, and simply share the link of the directory to us?

Comment (WangXin93) [reply configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/resnet_deepfashion2.md:55]: Ok, I will do that

Comment (WangXin93) [reply configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/resnet_deepfashion2.md]: I put all log in and models in this [link](https://drive.google.com/drive/folders/1AfYT_NOpBXHYmdDKjKjXcIJj8rYz_Ylb), it can also be found in the commit.

Comment (jin-s13) [reply configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/resnet_deepfashion2.md]: One more thing left, we will use this [publish_model tool](https://github.com/open-mmlab/mmpose/blob/master/tools/misc/publish_model.py) to rename the model ckpts before release. 
Could you help run this tool and follow our naming rules to rename the model ckpts and logs?

For example:
ckpt name:  res50_deepfashion_upper_256x192-41794f03_20210124.pth
log name: res50_deepfashion_upper_256x192_20210124.log.json

`res50_deepfashion_upper_256x192` is the same as the config .py
`41794f03` is sha256sum automatically generated by the publish_model tool.
`20210124` is the date automatically generated by the publish_model tool.

reference: https://raw.githubusercontent.com/open-mmlab/mmpose/master/configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion/resnet_deepfashion.md

Comment (WangXin93) [reply configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/resnet_deepfashion2.md]: > One more thing left, we will use this [publish_model tool](https://github.com/open-mmlab/mmpose/blob/master/tools/misc/publish_model.py) to rename the model ckpts before release. Could you help run this tool and follow our naming rules to rename the model ckpts and logs?
> 
> For example: ckpt name: res50_deepfashion_upper_256x192-41794f03_20210124.pth log name: res50_deepfashion_upper_256x192_20210124.log.json
> 
> `res50_deepfashion_upper_256x192` is the same as the config .py `41794f03` is sha256sum automatically generated by the publish_model tool. `20210124` is the date automatically generated by the publish_model tool.
> 
> reference: https://raw.githubusercontent.com/open-mmlab/mmpose/master/configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion/resnet_deepfashion.md

I will try

Comment (jin-s13) [reply configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/resnet_deepfashion2.md]: usage:

`python tools/misc/publish_model.py work_dirs/res50_deepfashion2_long_sleeved_dress_256x192/best_PCK_epoch_190.pth res50_deepfashion2_long_sleeved_dress_256x192`

Comment (WangXin93) [reply configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/resnet_deepfashion2.md]: I have uploaded renamed models to google drive. All links has been changed to the directory link for convenience.

Comment (jin-s13) [reply configs/fashion/2d_kpt_sview_rgb_img/topdown_heatmap/deepfashion2/resnet_deepfashion2.md]: @WangXin93  Thanks a lot! The PR looks good to me. We will merge it soon.

@ly015 Could you help make a backup copy of these model ckpts and logs in our openmmlab's cloud storage space?

**Review by ly015 — APPROVED** [review]

**Review by jin-s13 — APPROVED** [review]
